### PR TITLE
remove warning by updating action versions

### PR DESCRIPTION
**Pull request by nabobalis**
You are using "pypa/gh-action-pypi-publish@master". 
The "master" branch of this project has been sunset and will not receive any updates, not even security bug fixes. 
Please, make sure to use a supported version. If you want to pin to v1 major version, use "pypa/gh-action-pypi-publish@release/v1". 
If you feel adventurous, you may opt to use use "pypa/gh-action-pypi-publish@unstable/v1" instead. 
A more general recommendation is to pin to exact tags or commit shas.

### CODE REVIEW

**Comment by Cadair** [comment]
Lgtm but i will let @ConorMacBride or @astrofrog  merge it from not-my-phone

**Comment by nabobalis** [comment]
I also updated the versions for other actions

File: .github/workflows/publish.yml [hunk 1]
```diff
@@ -112,26 +112,26 @@ jobs:
...
           submodules: ${{ inputs.submodules }}
       - name: Set up QEMU
         if: ${{ matrix.CIBW_ARCHS == 'aarch64' }}
-        uses: docker/setup-qemu-action@v1
+        uses: docker/setup-qemu-action@v2
         with:
           platforms: all
       - name: Run cibuildwheel
-        uses: pypa/cibuildwheel@v2.5.0
+        uses: pypa/cibuildwheel@v2
```
Comment (ConorMacBride) [inline .github/workflows/publish.yml:126]: Don't think this will work as they don't track `v2` on the latest `v2._._` tag.

**Comment by nabobalis** [comment]
Uh oh it broke

File: .github/workflows/publish.yml (hunk 1 above)
Comment (nabobalis) [reply .github/workflows/publish.yml]: Changed.

**Comment by Cadair** [comment]
Can we run dependabot on this repo to keep these up to date?

**Comment by ConorMacBride** [comment]
> Can we run dependabot on this repo to keep these up to date?

See https://github.com/OpenAstronomy/github-actions-workflows/pull/69 and https://github.com/OpenAstronomy/build-python-dist/pull/14

**Pull request merged**